sklearn/metrics/tests/test_dist_metrics.py

```diff
@@ -370,7 +375,7 @@ def test_readonly_kwargs():
                 TypeError,
                 "Sparse data was passed for w, but dense data is required",
             )
-            for csr_container in CSR_CONTAINERS
+            for csr_container in COO_CONTAINERS
```

Comment:
Given that the test is checking for an error, I prefer updating the container to use 2d data for `w`.

(I did not know 1d SciPy CSR sparse matrices were a thing).